src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const newStyles = sourceString.trim();
+  const stylesArray = newStyles.split(';');
+  const arr = [];
+
+  for (let i = 0; i < stylesArray.length; i++) {
+    if (stylesArray[i].trim().length !== 0) {
+      arr.push(stylesArray[i].replace('\n', ''));
+    }
+  }
+
+  const resultArray = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    const part = arr[i].split(': ');
+
+    resultArray.push(part);
+  }
+
+  const result = resultArray.reduce((prev, item) => {
+    return {
+      ...prev,
+      [item[0].trim()]: item[1].trim(),
+    };
+  }, {});
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. These methods return a new array, so you should use it. In this case, `reduce` is used correctly to transform an array into an object. However, pay attention to the unnecessary curly braces and return statement inside the `reduce` callback, which can be omitted for a cleaner arrow function syntax.